core/unit_test/CMakeLists.txt

```diff
@@ -293,5 +294,61 @@ KOKKOS_ADD_EXECUTABLE_AND_TEST(
   SOURCES UnitTestMain.cpp  TestHostBarrier.cpp
 )
 
+FUNCTION (TEST_EXCLUDED CURRENT_TEST TEST_EXCLUDE_LIST)
+  FOREACH ( EXCLUDED_TEST ${TEST_EXCLUDE_LIST})
+    if(CURRENT_TEST STREQUAL ${EXCLUDED_TEST} )
+      SET (TEST_IMPLEMENTED OFF PARENT_SCOPE )
+    endif()
+  ENDFOREACH()
+ENDFUNCTION()
+
+
+FUNCTION (KOKKOS_ADD_INCREMENTAL_TEST BACK_END_NAME)
+
+  # Add unit test main
+  SET(${BACK_END_NAME}_SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/UnitTestMainInit.cpp)
+  
+  # Iterate over incremental tests in directory
+  APPEND_GLOB(INCREMENTAL_FILE_LIST ${CMAKE_CURRENT_SOURCE_DIR}/incremental/*.hpp)
+  foreach (CURRENT_FILE_PATH ${INCREMENTAL_FILE_LIST})
+    get_filename_component( CURRENT_FILE_NAME ${CURRENT_FILE_PATH} NAME )
+    string (REPLACE ".hpp" "" CURRENT_TEST_NAME ${CURRENT_FILE_NAME})
+    message( STATUS "OUT:${CURRENT_TEST_NAME}" )
+    SET ( TEST_IMPLEMENTED ON )
+    if (Kokkos_${BACK_END_NAME}_EXCLUDE_TESTS)
+       TEST_EXCLUDED( ${CURRENT_TEST_NAME} ${Kokkos_${BACK_END_NAME}_EXCLUDE_TESTS} )
```

Comment:
You don't need a function for that.  Just do
```cmake
if(NOT CURRENT_TEST_NAME IN_LIST Kokkos_${BACK_END_NAME}_EXCLUDE_TESTS)
  ...
endif()
```

Reply:
Fixed this..
Also added Kokkos_${BACK_END_NAME}_EXCLUDE_TESTS and Kokkos_${BACK_END_NAME}_FEATURE_LEVEL to CMakeCache using KOKKOS_OPTION